features/steps/test_update_resource.py

```diff
@@ -0,0 +1,239 @@
+    )
+    connection = engine.connect()
+    session_factory = sessionmaker(bind=connection)
+    Base.metadata.drop_all(engine)
+    Base.metadata.create_all(engine)
+
+    gateway = OcflRepositoryGateway(storage_path=path_data.storage)
+
+    yield SqlalchemyUnitOfWork(gateway=gateway, session_factory=session_factory)
+
+    connection.close()
+
+@pytest.fixture
+def message_bus(path_data: PathData, unit_of_work: AbstractUnitOfWork) -> MemoryMessageBus:
+    translocator = Translocator(
+        inbox_path=path_data.inbox,
+        workspaces_path=path_data.workspaces,
+        minter_provider=UuidMinterProvider(),
+        file_provider=FilesystemFileProvider()
+    )
+
+    handlers: dict[Type[Event], list[Callable]] = {
+        PackageSubmitted: [
+            lambda event: record_workflow_event(event, unit_of_work),
+            lambda event: receive_package(event, unit_of_work, translocator)
+        ],
+        PackageReceived: [
+            lambda event: record_workflow_event(event, unit_of_work),
+            lambda event: verify_package(event, unit_of_work, BagAdapter, Workspace, FilesystemFileProvider())
+        ],
+        PackageVerified: [
+            lambda event: record_workflow_event(event, unit_of_work),
+            lambda event: unpack_package(
+                event,
+                unit_of_work,
+                BagAdapter,
+                PackageResourceProvider,
+                Workspace,
+                FilesystemFileProvider(),
+            )
+        ],
+        PackageUnpacked: [
+            lambda event: record_workflow_event(event, unit_of_work),
+            lambda event: store_files(event, unit_of_work, Workspace)
+        ],
+        PackageStored: [
+            lambda event: record_workflow_event(event, unit_of_work),
+            lambda event: catalog_revision(event, unit_of_work)
+        ],
+        RevisionCataloged: [
+            lambda event: record_workflow_event(event, unit_of_work),
+        ],
+    }
+    message_bus = MemoryMessageBus(handlers)
+    return message_bus
+
+
+scenario = partial(scenario, '../update_resource.feature')
+
+@scenario('Updating a resource for immediate release')
+def test_updating_a_resource_for_immediate_release():
+    """Updating a resource for immediate release."""
+
+
+@given('a package containing all the scanned pages, OCR, and metadata')
+def _(path_data: PathData, unit_of_work: AbstractUnitOfWork, message_bus: MemoryMessageBus):
+    """a package containing all the scanned pages, OCR, and metadata."""
+    file_provider = FilesystemFileProvider()
+    file_provider.delete_dir_and_contents(path=path_data.scratch)
+    file_provider.create_directory(path_data.scratch)
+    file_provider.create_directory(path_data.storage)
+    file_provider.create_directory(path_data.workspaces)
+
+    unit_of_work.gateway.create_repository()
+
+    submission_id = "xyzzy-00000000-0000-0000-0000-000000000001-v1"
+    tracking_identifier = "first-load" # str(uuid.uuid4())
```

Comment:
If we want to do it this way, should we remove the comments here and below? I could also see just generating a couple UUIDs.